rust/agama-server/src/l10n/web.rs

```diff
@@ -119,14 +104,14 @@ async fn keymaps(State(state): State<LocaleState>) -> Json<Vec<Keymap>> {
 async fn set_config(
     State(state): State<LocaleState>,
     Json(value): Json<LocaleConfig>,
-) -> Result<Json<()>, LocaleError> {
+) -> Result<Json<()>, Error> {
```

Comment:
generic comment for whole this PR.
1. I like reduction of anyhow errors
2. I do not like this change in method signature. I think it is better to return specific error and use upper `Error` only when it can return more types of error.

Reply:
However, if you prefer, I could keep the `LocaleError` and implement `IntoResponse` only for that kind.

Reply:
you cannot have IntoResponse for whole Enum? If it is implemented for Error, I kind of expect it can process also element of Enum.

Reply:
or maybe some kind of trait can help there?

Reply:
It is implemented for the whole enum (the [crate::error::Error](https://github.com/openSUSE/agama/blob/better-errors/rust/agama-server/src/error.rs#L12)) and that's the reason I am returning `Error` instead of `LocaleError`. After all, `LocaleError` knows nothing about `Error`.

Adding any trait is overcomplicating things, in my opinion. So I would go for:

1) keep it as it is (returning `Error`) or...
2) returning `LocaleError` and implementing `IntoResponse`.

Reply:
Well, reason why I do not like it is because now it really see only use case for respond, but if it will be used in other parts of rust, then user of this function needs to handle any generic error, while we know that it can product only LocaleError.
So how I do see it is, that if we want to keep extensibility and allow in future to other types of Error in this function, lets change it to Error. But if we know that we always produce only LocaleError ( which I kind of expect ), then I prefer to return minimal abstraction it uses.

But ( and it is big BUT ) I do not care much in web functions that is short and really serve only as connection to real code and do some error translation. But this specific function is quite big and I would say many of its part is not web specific. So maybe if it lives elsewhere and this web function just call other method and translate its errortype into Error. Then it would much nicer.

And to conclude it, I still see this PR as improvement, so feel free to merge it.

Reply:
Yes, I [fully agree](https://github.com/openSUSE/agama/pull/1095#discussion_r1526457385), this code does not belong here but it is a step in the right direction.

I have added a note to [this Trello card](https://trello.com/c/npg6apH0/351-make-l10n-http-json-and-d-bus-interfaces-share-the-same-state) so it can be addressed later.